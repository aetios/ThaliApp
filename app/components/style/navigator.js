import { colors, create } from '../../style';

import { STATUSBAR_HEIGHT } from './standardHeader';

const styles = create({
  flex: {
    flex: 1,
  },
  statusBar: {
    height: STATUSBAR_HEIGHT,
    backgroundColor: colors.magenta,
  },
});

export default styles;
