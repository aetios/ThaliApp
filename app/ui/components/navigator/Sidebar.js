import React from 'react';
import PropTypes from 'prop-types';
import { Alert, Text, View, Image, TouchableHighlight, ImageBackground } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';
import Icon from 'react-native-vector-icons/MaterialIcons';
import styles from './style/Sidebar';

import * as navigationActions from '../../../actions/navigation';
import * as loginActions from '../../../actions/login';
import * as profileActions from '../../../actions/profile';
import Colors from '../../style/Colors';

const background = require('../../../assets/img/huygens.jpg');

const logoutPrompt = props => () => Alert.alert(
  props.t('Log out?'),
  props.t('Are you sure you want to log out?'),
  [{ text: props.t('No') },
    { text: props.t('Yes'), onPress: props.logout },
  ],
);

const Sidebar = (props) => {
  const buttons = [
    {
      onPress: () => props.navigate('welcome', true),
      iconName: 'home',
      text: props.t('Welcome'),
      style: {},
      scene: 'welcome',
    },
    {
      onPress: () => props.navigate('eventList', true),
      iconName: 'event',
      text: props.t('Calendar'),
      style: {},
      scene: 'eventList',
    },
    {
      onPress: () => props.navigate('members', true),
      iconName: 'people',
      text: props.t('Member List'),
      style: {},
      scene: 'members',
    },
    {
      onPress: logoutPrompt(props),
      iconName: 'lock',
      text: props.t('Logout'),
      style: {
        borderTopColor: Colors.lightGray,
        borderTopWidth: 1,
      },
      scene: 'logout',
    },
  ];

  return (
    <View
      style={styles.sidebar}
    >
      <TouchableHighlight
        onPress={() => props.loadProfile(props.token)}
        style={styles.headerButton}
      >
        <ImageBackground
          source={background}
          style={styles.headerImage}
          resizeMode="cover"
        >
          <LinearGradient colors={['#55000000', '#000000']} style={styles.headerGradient} />
          <Image
            source={{ uri: props.photo }}
            style={styles.profileImage}
            resizeMode="cover"
          />
          <Text style={styles.nameField}>{props.displayName}</Text>
        </ImageBackground>
      </TouchableHighlight>
      <View style={styles.buttonList}>
        {buttons.map(button => (
          <Icon.Button
            onPress={button.onPress}
            name={button.iconName}
            borderRadius={0}
            backgroundColor={Colors.white}
            color={props.currentScene === button.scene ?
                 Colors.magenta : Colors.textColour}
            size={24}
            iconStyle={styles.buttonIcon}
            style={[styles.buttonText, button.style]}
            key={button.scene}
          >
            {button.text}
          </Icon.Button>
        ))}
      </View>
    </View>
  );
};

Sidebar.propTypes = {
  currentScene: PropTypes.string.isRequired,
  displayName: PropTypes.string.isRequired,
  photo: PropTypes.string.isRequired,
  token: PropTypes.string.isRequired,
  // eslint-disable-next-line react/no-unused-prop-types
  logout: PropTypes.func.isRequired,
  loadProfile: PropTypes.func.isRequired,
  // eslint-disable-next-line react/no-unused-prop-types
  navigate: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  currentScene: state.navigation.currentScene,
  displayName: state.session.displayName,
  photo: state.session.photo,
  token: state.session.token,
});

const mapDispatchToProps = dispatch => ({
  navigate: (scene, newSection = false) => dispatch(navigationActions.navigate(scene, newSection)),
  logout: () => dispatch(loginActions.logout()),
  loadProfile: token => dispatch(profileActions.profile(token)),
});

export default connect(mapStateToProps, mapDispatchToProps)(translate('components/navigator/Sidebar')(Sidebar));
