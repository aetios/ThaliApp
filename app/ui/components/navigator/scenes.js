export const SPLASH_SCENE = 'splash';
export const WELCOME_SCENE = 'welcome';
export const EVENT_SCENE = 'event';
export const EVENT_LIST_SCENE = 'eventList';
export const PROFILE_SCENE = 'profile';
export const PIZZA_SCENE = 'pizza';
export const REGISTRATION_SCENE = 'registration';
export const MEMBERS_SCENE = 'members';
export const SETTINGS_SCENE = 'settings';
export const LOGIN_SCENE = 'login';
