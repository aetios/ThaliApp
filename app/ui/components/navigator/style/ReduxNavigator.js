import Colors from '../../../style/Colors';
import StyleSheet from '../../../style/StyleSheet';

import { STATUSBAR_HEIGHT } from '../../standardHeader/style/StandardHeader';

const styles = StyleSheet.create({
  flex: {
    flex: 1,
  },
  statusBar: {
    height: STATUSBAR_HEIGHT,
    backgroundColor: Colors.magenta,
  },
});

export default styles;
