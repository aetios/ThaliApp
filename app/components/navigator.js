import React from 'react';
import PropTypes from 'prop-types';
import { View, StatusBar, BackHandler } from 'react-native';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';
import Drawer from 'react-native-drawer';
import Login from './Login';
import Welcome from './Welcome';
import Sidebar from './Sidebar';
import Event from './Event';
import Calendar from './Calendar';
import Profile from './Profile';
import Pizza from './Pizza';
import StandardHeader from './StandardHeader';
import Registration from './Registration';
import MemberList from './MemberList';

import * as actions from '../actions/navigation';
import styles from './style/navigator';
import { colors } from '../style';

const sceneToComponent = (scene) => {
  switch (scene) {
    case 'welcome':
      return <Welcome />;
    case 'event':
      return <Event />;
    case 'eventList':
      return <Calendar />;
    case 'profile':
      return <Profile />;
    case 'pizza':
      return <Pizza />;
    case 'registration':
      return <Registration />;
    case 'members':
      return <MemberList />;
    default:
      return <Welcome />;
  }
};

const ReduxNavigator = (props) => {
  const { currentScene, loggedIn, drawerOpen, updateDrawer,
          isFirstScene, back, navigateToWelcome } = props;
  BackHandler.addEventListener('hardwareBackPress', () => {
    if (!isFirstScene) {
      back();
      return true;
    } else if (currentScene !== 'welcome') {
      navigateToWelcome();
      return true;
    }
    BackHandler.exitApp();
    return true;
  });
  if (loggedIn) {
    return (<Drawer
      type="overlay"
      content={<Sidebar />}
      openDrawerOffset={0.2}
      panOpenMask={0.2}
      panCloseMask={0.2}
      panThreshold={0.3}
      styles={{
        mainOverlay: {
          backgroundColor: colors.black,
          opacity: 0,
          elevation: 100,
        },
      }}
      tweenHandler={ratio => ({ mainOverlay: { opacity: ratio * 0.75 } })}
      open={drawerOpen}
      onOpen={() => updateDrawer(true)}
      onClose={() => updateDrawer(false)}
      tapToClose
    >
      {currentScene !== 'profile' && currentScene !== 'members' && <StandardHeader />}
      {sceneToComponent(currentScene)}
    </Drawer>);
  }
  return (
    <View
      style={styles.flex}
    >
      <View style={styles.statusBar}>
        <StatusBar
          backgroundColor={colors.semiTransparent}
          barStyle="light-content"
          translucent
          animated
        />
      </View>
      <Login />
    </View>);
};

ReduxNavigator.propTypes = {
  currentScene: PropTypes.string.isRequired,
  loggedIn: PropTypes.bool.isRequired,
  drawerOpen: PropTypes.bool.isRequired,
  isFirstScene: PropTypes.bool.isRequired,
  updateDrawer: PropTypes.func.isRequired,
  back: PropTypes.func.isRequired,
  navigateToWelcome: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  currentScene: state.navigation.currentScene,
  loggedIn: state.navigation.loggedIn,
  drawerOpen: state.navigation.drawerOpen,
  isFirstScene: state.navigation.previousScenes.length === 0,
});

const mapDispatchToProps = dispatch => ({
  updateDrawer: isOpen => dispatch(actions.updateDrawer(isOpen)),
  back: () => dispatch(actions.back()),
  navigateToWelcome: () => dispatch(actions.navigate('welcome', true)),
});

export default connect(mapStateToProps, mapDispatchToProps)(translate('navigator')(ReduxNavigator));
