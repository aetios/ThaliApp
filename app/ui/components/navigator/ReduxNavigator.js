import React from 'react';
import PropTypes from 'prop-types';
import { View, StatusBar, BackHandler } from 'react-native';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';
import Drawer from 'react-native-drawer';
import Login from '../../screens/user/Login';
import Welcome from '../../screens/welcome/Welcome';
import Sidebar from './Sidebar';

import Event from '../../screens/events/Event';
import Calendar from '../../screens/events/Calendar';
import Profile from '../../screens/user/Profile';
import Pizza from '../../screens/pizza/Pizza';
import StandardHeader from '../standardHeader/StandardHeader';
import Registration from '../../screens/events/Registration';
import MemberList from '../../screens/memberList/MemberList';
import Settings from '../../screens/settings/Settings';

import * as actions from '../../../actions/navigation';
import styles from './style/ReduxNavigator';
import Colors from '../../style/Colors';

const sceneToComponent = (scene) => {
  switch (scene) {
    case 'welcome':
      return <Welcome />;
    case 'event':
      return <Event />;
    case 'eventList':
      return <Calendar />;
    case 'profile':
      return <Profile />;
    case 'pizza':
      return <Pizza />;
    case 'registration':
      return <Registration />;
    case 'members':
      return <MemberList />;
    case 'settings':
      return <Settings />;
    default:
      return <Welcome />;
  }
};

const ReduxNavigator = (props) => {
  const {
    currentScene, loggedIn, drawerOpen, updateDrawer,
    isFirstScene, back, navigateToWelcome,
  } = props;
  BackHandler.addEventListener('hardwareBackPress', () => {
    if (!isFirstScene) {
      back();
      return true;
    } if (currentScene !== 'welcome') {
      navigateToWelcome();
      return true;
    }
    BackHandler.exitApp();
    return true;
  });
  if (loggedIn) {
    return (
      <Drawer
        type="overlay"
        content={<Sidebar />}
        openDrawerOffset={0.2}
        panOpenMask={0.2}
        panCloseMask={0.2}
        panThreshold={0.3}
        styles={{
          mainOverlay: {
            backgroundColor: Colors.black,
            opacity: 0,
            elevation: 100,
          },
        }}
        tweenHandler={ratio => ({ mainOverlay: { opacity: ratio * 0.75 } })}
        open={drawerOpen}
        onOpen={() => updateDrawer(true)}
        onClose={() => updateDrawer(false)}
        tapToClose
      >
        {currentScene !== 'profile' && currentScene !== 'members' && <StandardHeader />}
        {sceneToComponent(currentScene)}
      </Drawer>
    );
  }
  return (
    <View
      style={styles.flex}
    >
      <View style={styles.statusBar}>
        <StatusBar
          backgroundColor={Colors.semiTransparent}
          barStyle="light-content"
          translucent
          animated
        />
      </View>
      <Login />
    </View>);
};

ReduxNavigator.propTypes = {
  currentScene: PropTypes.string.isRequired,
  loggedIn: PropTypes.bool.isRequired,
  drawerOpen: PropTypes.bool.isRequired,
  isFirstScene: PropTypes.bool.isRequired,
  updateDrawer: PropTypes.func.isRequired,
  back: PropTypes.func.isRequired,
  navigateToWelcome: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  currentScene: state.navigation.currentScene,
  loggedIn: state.navigation.loggedIn,
  drawerOpen: state.navigation.drawerOpen,
  isFirstScene: state.navigation.previousScenes.length === 0,
});

const mapDispatchToProps = dispatch => ({
  updateDrawer: isOpen => dispatch(actions.updateDrawer(isOpen)),
  back: () => dispatch(actions.back()),
  navigateToWelcome: () => dispatch(actions.navigate('welcome', true)),
});

export default connect(mapStateToProps, mapDispatchToProps)(translate('components/navigator/ReduxNavigator')(ReduxNavigator));
