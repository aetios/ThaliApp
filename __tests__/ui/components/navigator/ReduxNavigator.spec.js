import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import ReduxNavigator from '../../../../app/ui/components/navigator/ReduxNavigator';
import reducer from '../../../../app/reducers';
import { LOGIN_SCENE } from '../../../../app/ui/components/navigator/scenes';

describe('ReduxNavigator component', () => {
  const mockStore = configureStore(reducer);
  const initialState = {
    navigation: {
      currentScene: LOGIN_SCENE,
      previousScenes: [],
      drawerOpen: false,
    },
  };
  const store = mockStore(initialState);

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ReduxNavigator />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
