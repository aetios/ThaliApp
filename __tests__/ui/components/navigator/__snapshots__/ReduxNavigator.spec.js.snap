// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReduxNavigator component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#E62272",
        "height": 20,
      }
    }
  />
  <View
    modalOpen="false"
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#E62272",
          "flex": 1,
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={false}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/img/logo.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "height": 108,
            "marginBottom": 48,
            "resizeMode": "contain",
            "width": 260,
          }
        }
      />
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Username"
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 4,
              "height": 48,
              "marginBottom": 8,
              "marginLeft": 16,
              "marginRight": 16,
              "padding": 8,
            }
          }
          underlineColorAndroid="#313131"
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 4,
              "height": 48,
              "marginBottom": 8,
              "marginLeft": 16,
              "marginRight": 16,
              "padding": 8,
            }
          }
          underlineColorAndroid="#313131"
        />
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "marginBottom": 16,
              "marginTop": 32,
              "width": 120,
            },
            Object {
              "borderRadius": 100,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 100,
                "padding": 10,
                "paddingLeft": 12,
                "paddingRight": 12,
              },
              Object {
                "backgroundColor": "#373737",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "justifyContent": "center",
                  "textAlign": "center",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            LOGIN
          </Text>
        </View>
      </View>
      <Text
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "color": "#373737",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "600",
            "height": 25,
            "marginBottom": 10,
          }
        }
      >
        Forgot password?
      </Text>
    </View>
  </View>
</View>
`;
